open-cas does not support Ubuntu 19.04 (kernel 5.0.0)

stack@ceph01liang: ~ / open-cas-linux$ sudo make install
cd modules && make install
make[1]: Entering directory '/opt/stack/open-cas-linux/modules'
Installing Open-CAS modules
modprobe: ERROR: could not insert 'cas_cache': Unknown symbol in module, or unknown parameter (see dmesg)
make[1]: *** [Makefile:58: install] Error 1
make[1]: Leaving directory '/opt/stack/open-cas-linux/modules'
make: *** [Makefile:15: modules] Error 2

stack@ceph01liang: ~ / open-cas-linux$ uname -r
5.0.0-29-generic


--------------------------------

stack@ceph01liang:~/open-cas-linux$ dmesg | grep cas_disk
[91506.368989] cas_disk: loading out-of-tree module taints kernel.
[91506.369108] cas_disk: module verification failed: signature and/or required key missing - tainting kernel

--- a/configure.d/1_blk_end_req.conf
+++ b/configure.d/1_blk_end_req.conf
@@ -9,20 +9,23 @@
 check() {
 	cur_name=$(basename $2)
 	config_file_path=$1
-	if compile_module $cur_name "blk_end_request_all(NULL, 0)" "linux/blkdev.h"
+	if compile_module $cur_name "blk_mq_end_request(NULL, 0)" "linux/blk-mq.h"
 	then
 		echo $cur_name "1" >> $config_file_path
-	else
+	elif compile_module $cur_name "blk_end_request_all(NULL, 0)" "linux/blkdev.h"
+	then
 		echo $cur_name "2" >> $config_file_path
-    fi
+	else
+		echo $cur_name "X" >> $config_file_path
+	fi
 }
 
 apply() {
     case "$1" in
     "1")
-		add_define "CAS_END_REQUEST_ALL blk_end_request_all" ;;
-    "2")
 		add_define "CAS_END_REQUEST_ALL blk_mq_end_request" ;;
+    "2")
+		add_define "CAS_END_REQUEST_ALL blk_end_request_all" ;;
     *)
         exit 1
     esac

--- a/modules/cas_cache/linux_kernel_version.h
+++ b/modules/cas_cache/linux_kernel_version.h
@@ -38,6 +38,7 @@
 #include <linux/nmi.h>
 #include <linux/ratelimit.h>
 #include <linux/mm.h>
+#include <linux/blk-mq.h>
 
 #include "generated_defines.h"
 

